examples/cli_parsing.rs

```diff
@@ -0,0 +1,48 @@
+#![feature(async_await, futures_api)]
+
+use structopt::StructOpt;
+use tide::{configuration::Configuration, ExtractConfiguration};
+
+#[derive(Debug, StructOpt)]
+#[structopt(name = "tide_example")]
+struct Settings {
+    #[structopt(
+        short = "a",
+        long = "address",
+        env = "TIDE_ADDRESS",
+        default_value = "127.0.0.1"
+    )]
+    address: String,
+    #[structopt(short = "p", long = "port", env = "TIDE_PORT", default_value = "8000")]
+    port: u16,
+    #[structopt(
+        short = "d",
+        long = "database",
+        env = "TIDE_DATABASE",
```

Comment:
I think as an example it's fine to drop the `TIDE_` prefix for these env vars. For example Heroku [always sets the PORT env var](https://stackoverflow.com/a/28707129/1541707). This might help clarify there's nothing special about how tide needs to interface with env vars.